sim/core/periodic_action.go

```diff
@@ -61,12 +52,15 @@ func NewPeriodicAction(sim *Simulation, options PeriodicActionOptions) *PendingA
 	pa := &PendingAction{
```

Comment:
... and here as well. This allows to keep the implicit "PendingAction.OnAction() must not be nil" invariant elsewhere, and makes it more explicit.
The code changes become very minimalistic then, too :)

Reply:
Done!